javascript/javascript1/week3/smartphoneusage.js

```diff
@@ -0,0 +1,61 @@
+const activities = [];
+const usageLimit = 120;
+function addActivity(activity, duration) {
+  activities.push({
+    date: new Date().toLocaleDateString("en-US"),
+    activity: activity,
+    duration: duration,
+  });
+}
+
+function showStatus(activities) {
+  if (activities.length > 0) {
+    let totalDuration = 0;
+    let countActivities = 0;
+    const today = new Date().toLocaleDateString("en-US");
+    for (obj of activities) {
+      if (obj.date === today) {
+        totalDuration += Number(obj.duration);
+        countActivities += 1;
+      }
+    }
+    const msg =
+      totalDuration > usageLimit
+        ? "You have reached your limit, no more smartphoning for you!"
+        : `You are ${
+            usageLimit - totalDuration
+          } min. to reach daily usage limit`;
+    console.log(
+      `You have added ${countActivities} activities on ${today}. They amount to ${totalDuration} min. of usage. ${msg}`
+    );
+  } else console.log("Add some activities before calling showStatus");
+}
+
+function mostTimeSpent(activities) {
```

Comment:
I don't understand what is happening here, but I can see that there are some confusions. 
1.`let newActivities` is the same as activities, we don't really need it.
2. `.reduce()` gets at least two parameters `accumulator` and `currentValue)` , you can also initial a value, but if not basically means the accumulator is `array[0]`. Here in line 39, you use `.reduce()` method for `newActivities`, the way that you are using it could be interpreted like `agg = newActivities[0]` and `acti = newActivities[1]`, so we can not write `agg[acti.activity]`,and this is where that I can not understand what is going on.